python/dgl/convert.py

```diff
@@ -750,19 +750,33 @@ def to_homo(G):
     nids = []
     total_num_nodes = 0
 
+    ntid_dict = G.ndata[NTYPE] if len(G.ntypes) > 1 else {}
     for ntype_id, ntype in enumerate(G.ntypes):
         num_nodes = G.number_of_nodes(ntype)
-        total_num_nodes += num_nodes
-        ntype_ids.append(F.full_1d(num_nodes, ntype_id, F.int64, F.cpu()))
+        if len(ntid_dict) > 0:
+            ntid = ntid_dict.get(ntype, None)
+            assert ntid is not None, 'if dgl.NTYPE field exist in heterograph, ' \
+                'all nodes should have this field.'
+            ntype_ids.append(ntid)
```

Comment:
It seems `num_nodes` is not used in this branch and `ntype_ids` in L760 has a different length compared to `ntype_ids` in L763, how do you address the discrepancy?